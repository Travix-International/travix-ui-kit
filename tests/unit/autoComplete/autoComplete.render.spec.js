import { shallow } from 'enzyme';
import React from 'react';
import AutoComplete from '../../../components/autoComplete/autoComplete';
import AutoCompleteItem from '../../../components/autoComplete/autoCompleteItem';

describe('AutoComplete', () => {
  describe('#render()', () => {
    it('should render correct autocomplete component', () => {
      const wrapper = shallow(
        <AutoComplete
          className="test-auto-complete"
          name="autocomplete"
        >
          <AutoCompleteItem
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should copy mods', () => {
      const mods = ['test'];
      const wrapper = shallow(
        <AutoComplete
          mods={mods}
          name="autocomplete"
        >
          <AutoCompleteItem
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render correct open autocomplete component', () => {
      const component = shallow(
        <AutoComplete
          name="autocomplete"
        >
          <AutoCompleteItem
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      component.setState({ open: true });

      expect(component).toMatchSnapshot();
    });

    it('should render correct autocomplete component with label', () => {
      const component = shallow(
        <AutoComplete
          label="label"
          name="autocomplete"
        >
          <AutoCompleteItem
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      expect(component).toMatchSnapshot();
    });

    it('should render correct autocomplete component with title item', () => {
      const component = shallow(
        <AutoComplete
          label="label"
          name="autocomplete"
        >
          <AutoCompleteItem
            isTitle
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      expect(component).toMatchSnapshot();
    });

    it('should render correct autocomplete component with highlighting', () => {
      const component = shallow(
        <AutoComplete
          highlighted
          name="autocomplete"
        >
          <AutoCompleteItem
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      expect(component).toMatchSnapshot();
    });

    it('should render correct autocomplete component with default value', () => {
      const component = shallow(
        <AutoComplete
          defaultValue={{ value: 'value' }}
          highlighted
          name="autocomplete"
        >
          <AutoCompleteItem
            value="value"
          >
            item
          </AutoCompleteItem>
        </AutoComplete>
      );

      expect(component).toMatchSnapshot();
    });

    it('should not render item child if it not "AutoCompleteItem"', () => {
      const component = shallow(
        <AutoComplete
          name="autocomplete"
        >
          <div/>
        </AutoComplete>
      );

      expect(component).toMatchSnapshot();
    });
  });
});
