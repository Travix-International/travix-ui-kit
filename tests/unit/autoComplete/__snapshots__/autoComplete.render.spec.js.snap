// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete #render() should copy mods 1`] = `
<div
  className="ui-autocomplete ui-autocomplete_test"
  highlighted={false}
  mods={
    Array [
      "test",
    ]
  }
>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby=""
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={true}
        isTitle={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        item
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;

exports[`AutoComplete #render() should not render item child if it not "AutoCompleteItem" 1`] = `
<div
  className="ui-autocomplete"
  highlighted={false}
>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby=""
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    />
  </div>
</div>
`;

exports[`AutoComplete #render() should render correct autocomplete component 1`] = `
<div
  className="ui-autocomplete"
  highlighted={false}
>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby=""
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={true}
        isTitle={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        item
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;

exports[`AutoComplete #render() should render correct autocomplete component with default value 1`] = `
<div
  className="ui-autocomplete"
  defaultValue={
    Object {
      "value": "value",
    }
  }
  highlighted={true}
>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby=""
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
    value="value"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={true}
        isTitle={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "item",
            }
          }
        />
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;

exports[`AutoComplete #render() should render correct autocomplete component with highlighting 1`] = `
<div
  className="ui-autocomplete"
  highlighted={true}
>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby=""
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={true}
        isTitle={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        item
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;

exports[`AutoComplete #render() should render correct autocomplete component with label 1`] = `
<div
  className="ui-autocomplete"
  highlighted={false}
>
  <label
    htmlFor="ui-autocomplete-input-autocomplete"
    id="ui-autocomplete-label-autocomplete"
  >
    label
  </label>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby="ui-autocomplete-label-autocomplete"
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={true}
        isTitle={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        item
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;

exports[`AutoComplete #render() should render correct autocomplete component with title item 1`] = `
<div
  className="ui-autocomplete"
  highlighted={false}
>
  <label
    htmlFor="ui-autocomplete-input-autocomplete"
    id="ui-autocomplete-label-autocomplete"
  >
    label
  </label>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-labelledby="ui-autocomplete-label-autocomplete"
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={false}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={false}
        isTitle={true}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        item
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;

exports[`AutoComplete #render() should render correct open autocomplete component 1`] = `
<div
  className="ui-autocomplete ui-autocomplete_open"
  highlighted={false}
>
  <Input
    aria-activedescendant="ui-autocomplete-item-autocomplete-0"
    aria-autocomplete="list"
    aria-expanded={true}
    aria-haspopup={true}
    aria-labelledby=""
    aria-owns="ui-autocomplete-list-autocomplete"
    autoComplete="off"
    disabled={false}
    id="ui-autocomplete-input-autocomplete"
    multiline={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="combobox"
  />
  <Input
    disabled={false}
    hidden={true}
    multiline={false}
    name="autocomplete"
  />
  <div
    className="ui-autocomplete__popunder"
  >
    <ul
      aria-expanded={true}
      className="ui-autocomplete__popunder-list"
      id="ui-autocomplete-list-autocomplete"
      role="listbox"
    >
      <AutoCompleteItem
        id="ui-autocomplete-item-autocomplete-0"
        index={0}
        isActive={true}
        isTitle={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        value="value"
      >
        item
      </AutoCompleteItem>
    </ul>
  </div>
</div>
`;
